src/clusterfuzz/_internal/metrics/monitor.py

```diff
@@ -108,13 +109,21 @@ def run(self):
                    f'metric_kind: {series.metric_kind}, '
                    f'start_time: {metric_st_sec}.{metric_st_ns}, '
                    f'end_time: {metric_et_sec}.{metric_et_ns}')
-          time_series.append(series)
+
+          if (series.metric_kind ==
+              monitoring_v3.enums.MetricDescriptor.MetricKind.GAUGE):
+            gauge_series.append(series)
+          else:
+            time_series.append(series)
 
           if len(time_series) == MAX_TIME_SERIES_PER_CALL:
+            time_series.extend(gauge_series)
             create_time_series(project_path, time_series)
+            gauge_series = []
             time_series = []
 
         if time_series:
+          time_series.extend(gauge_series)
           create_time_series(project_path, time_series)
```

Comment:
Did you have the chance to run this locally and verify that the data structure `monitoring_v3.enums.MetricDescriptor.MetricKind.GAUGE` is compatible downstream in `create_time_series`?

I just want to make sure this won't cause any unexpected issues when creating the time series.

It may make sense to add or modify the tests in `src/clusterfuzz/_internal/tests/core/metrics/monitor_test.py` to verify this

Reply:
Hi Mark,
Replaced monitoring_v3.enums.MetricDescriptor.MetricKind.GAUGE with metric_pb2.MetricDescriptor.MetricKind.GAUGE, and it's now working fine. However, when verifying whether our changes are compatible downstream in create_time_series, I'm encountering a "Permission Denied" issue when trying to call create_time_series.
 
This change sorts the existing series which includes the GAUGE metric in an ascending order of start_time. No new type of metrics has been added. Hence the existing MetricKind.GAUGE should be compatible with downstream in create_time_series.
 
Additionally, added a test for GaugeMetrics to monitor_test.py. Our changes include logical modifications to the FlusherThread.run() API. However, the corresponding test is marked with @unittest.skip, so it does not actually test our changes.

Reply:
Ack, thanks for adding the test as well.